dmoj/executors/__init__.py

```diff
@@ -15,6 +15,28 @@
 executors: Dict[str, Any] = {}
 
 
+def by_ext(ext : str):
+    ext = ext.lower()
+
+    for name, executor in executors.items():
+        if name.lower() == ext:
+            return executor
+
+    for executor in executors.values():
+        if executor.ext == ext:
```

Comment:
Is this deterministic? I really don't want it to flip randomly between python 2 and python 3.

Reply:
Added a sort to be safe.